lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -48,6 +48,9 @@ def find(request)
     node.trusted_data = Puppet.lookup(:trusted_information) { Puppet::Context::TrustedInformation.local(node) }.to_h
 
     if catalog = compile(node)
+      if code_id = request.options[:code_id]
```

Comment:
Sorry for not seeing this earlier: it looks like with this setup, `code_id` is not available during compilation, though we need that so we can query PDB with it when we look up consumed resources.